src/formatting_stack/branch_formatter.clj

```diff
@@ -75,37 +81,60 @@
                                 strategies/jvm-requirable-files
                                 strategies/namespaces-within-refresh-dirs-only)))])
 
-(def default-processors
+(speced/defn default-processors [_
+                                 ^vector? processors-strategies
+                                 ^map? third-party-indent-specs]
   [(processors.cider/new {:third-party-indent-specs third-party-indent-specs})])
 
 (def default-reporter
   (pretty-printer/new {}))
 
-(defn format-and-lint-branch! [& {:keys [target-branch in-background? reporter]
-                                  :or   {target-branch  "master"
-                                         in-background? (not (System/getenv "CI"))
-                                         reporter       default-reporter}}]
+(def formatter-factory
+  (implement {}
+    protocols.formatter/--formatters default-formatters))
+
+(def no-formatters (constantly []))
+
+(def empty-formatter-factory
+  (implement {}
+    protocols.formatter/--formatters no-formatters))
+
+(def linter-factory
+  (implement {}
+    protocols.linter/--linters default-linters))
+
+(def processor-factory
+  (implement {}
+    protocols.processor/--processors default-processors))
+
+(defn format-and-lint-branch!
+  "Note that files that are not completely staged will not be affected.
+
+  You can override that with `:blacklist []`."
```

Comment:
The docstring is correctly reflects the intended behavior, however not _linting_ unstaged files is suboptimal.

After all, linting things is safe, so there's no reason to omit it (and it tends to be surprising; has bit me in the past).

Should be easy to fix.